packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -111,57 +113,52 @@ class UserSessionManager implements IUserSessionManager {
       this.setInitialReferringDomain(getReferringDomain(referrer));
     }
     // Initialize session tracking
-    if (this.isStorageTypeValidForStoringData(sessionTrackingStorage)) {
+    if (this.isPersistanceEnabledForStorageEntry(UserSessionKeys.sessionInfo)) {
       this.initializeSessionTracking();
-    } else {
-      state.session.sessionInfo.value = defaultUserSessionValues.sessionInfo;
     }
     // Register the effect to sync with storage
     this.registerEffects();
   }
 
-  isStorageTypeValidForStoringData(storage: StorageType): boolean {
-    return !!(
-      storage === COOKIE_STORAGE ||
-      storage === LOCAL_STORAGE ||
-      storage === MEMORY_STORAGE
-    );
+  isPersistanceEnabledForStorageEntry(entryName: UserSessionKeysType): boolean {
+    const entries = state.storage.entries.value;
+    return isStorageTypeValidForStoringData(entries[entryName]?.type as StorageType);
   }
 
   migrateDataFromPreviousStorage() {
-    const entries: UserSessionEntries = state.storage.entries.value as UserSessionEntries;
+    const entries = state.storage.entries.value as StorageEntries;
     Object.keys(entries).forEach(entry => {
       const key = entry as UserSessionStorageKeysType;
-      const currentStorage = entries[key].storage;
-      const storages: MigrationStorageType[] = [COOKIE_STORAGE, LOCAL_STORAGE];
+      const currentStorage = entries[key]?.type as StorageType;
+      const curStore = this.storeManager?.getStore(storageClientDataStoreNameMap[currentStorage]);
+      const storages = [COOKIE_STORAGE, LOCAL_STORAGE];
 
       storages.forEach(storage => {
         const store = this.storeManager?.getStore(storageClientDataStoreNameMap[storage]);
-        const curStore = this.storeManager?.getStore(storageClientDataStoreNameMap[currentStorage]);
-        if (
-          storage !== currentStorage &&
-          store &&
-          this.isStorageTypeValidForStoringData(currentStorage)
-        ) {
-          const value = store?.get(userSessionStorageKeys[key]);
-          if (value) {
-            if (curStore) {
+        if (storage !== currentStorage && store) {
+          if (curStore) {
+            const value = store?.get(userSessionStorageKeys[key]);
+            if (value) {
               curStore?.set(userSessionStorageKeys[key], value);
             }
-            store?.remove(userSessionStorageKeys[key]);
           }
+          store?.remove(userSessionStorageKeys[key]);
         }
       });
     });
   }
 
   migrateStorageIfNeeded() {
+    if (!state.storage.migrate.value) {
+      return;
+    }
     const cookieStorage = this.storeManager?.getStore(CLIENT_DATA_STORE_COOKIE);
     const localStorage = this.storeManager?.getStore(CLIENT_DATA_STORE_LS);
```

Comment:
You can put these into an array and loop over to avoid code duplication below.

Reply:
Addressed